File: 14-longest-common-prefix/14-longest-common-prefix.cpp
class Solution {
public:
    bool checkAllEqual(vector<string>& strs,int idx){
        auto reference = strs[0][idx];
        for(auto s:strs){
            if(s[idx]!=reference) return false;
        }
        return true;
    }
    
    string longestCommonPrefix(vector<string>& strs) {
        string ans="";
        unsigned long minimum_length=INT_MAX;
        for(auto i : strs){
            minimum_length = min(minimum_length,i.size());
        }
        auto length=0;
        for(;length<minimum_length;length++){
            if(!checkAllEqual(strs,length)){
                break;
            }
            else{
                ans+=strs[0][length];
            }
        }
        
        return ans;
    }
};

File: 14-longest-common-prefix/14-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        N = len(strs)
        idx = 0
        while True:
            char = None
            for i in range(N):
                if len(strs[i])<=idx:
                    return strs[i][:idx]
                else:
                    if char==None:
                        char = strs[i][idx]
                    else:
                        if char!=strs[i][idx]:return strs[i][:idx]
            idx+=1
